assets/src/design-system/components/contextMenu/animatedContextMenu.js

```diff
@@ -0,0 +1,287 @@
+      props.isReady ? 'none' : `scale(${initialScale})`};
+    transform-origin: ${(props) => {
+      switch (props.align) {
+        case CORNER_DIRECTIONS.top_right:
+          return 'right top';
+        case CORNER_DIRECTIONS.top_left:
+          return 'left top';
+        case CORNER_DIRECTIONS.bottom_right:
+          return 'right bottom';
+        case CORNER_DIRECTIONS.bottom_left:
+        default:
+          return 'left bottom';
+      }
+    }};
+  }
+`;
+ButtonInner.propTypes = {
+  align: PropTypes.oneOf(Object.values(CORNER_DIRECTIONS)),
+  isReady: PropTypes.bool,
+};
+
+function AnimationContainer({ children, isOpen, ...props }) {
+  const [align, setAlign] = useState(null);
+  const [isReady, setIsReady] = useState(false);
+  const menuPositionRef = useRef(null);
+  const menuTogglePositionRef = useRef(null);
+
+  useLayoutEffect(() => {
+    if (!isOpen) {
+      setAlign(null);
+      return;
+    }
+    if (!(menuTogglePositionRef.current && menuPositionRef.current)) {
+      return;
+    }
+
+    const toggleBoundingBox = menuTogglePositionRef.current.getBoundingClientRect();
+    const menuWrapperBoundingBox = menuPositionRef.current.getBoundingClientRect();
+
+    const menuLeft =
+      toggleBoundingBox.left +
+      toggleBoundingBox.width * (PERCENTAGE_OFFSET[DIRECTIONS.LEFT] / 100);
+    const menuBottom =
+      toggleBoundingBox.bottom +
+      toggleBoundingBox.height * (PERCENTAGE_OFFSET[DIRECTIONS.BOTTOM] / 100);
+
+    const alignHorizontal =
+      menuLeft + menuWrapperBoundingBox.width > window.innerWidth
+        ? DIRECTIONS.RIGHT
+        : DIRECTIONS.LEFT;
+    const alignVertical =
+      0 > menuBottom - menuWrapperBoundingBox.height
+        ? DIRECTIONS.TOP
+        : DIRECTIONS.BOTTOM;
+
+    setAlign(CORNER_DIRECTIONS[`${alignVertical}_${alignHorizontal}`]);
+  }, [isOpen]);
+
+  /**
+   * Seems funky, but we need 1 full render where the proper
+   * alignment is set before we animate in. This prevents react
+   * from batching those renders and animating from wrong alignemnt.
+   */
+  useEffect(() => {
+    const frameId = requestAnimationFrame(() => setIsReady(Boolean(align)));
+
+    return () => cancelAnimationFrame(frameId);
+  }, [align]);
+
+  return (
+    <ButtonInner
+      align={align}
+      isOpen={isOpen}
+      isReady={isReady}
+      ref={menuTogglePositionRef}
+      {...props}
+    >
+      <MenuWrapper>
+        <MenuRevealer>
+          <MenuCounterRevealer ref={menuPositionRef}>
+            {children}
+          </MenuCounterRevealer>
+        </MenuRevealer>
+        <Shadow />
+      </MenuWrapper>
+    </ButtonInner>
+  );
+}
+AnimationContainer.propTypes = {
+  isOpen: PropTypes.bool,
+  children: PropTypes.node,
+};
+
+const AnimatedContextMenu = ({ isOpen, items }) => {
+  return (
+    <AnimationContainer isOpen={isOpen}>
+      <Menu items={items} />
+    </AnimationContainer>
+  );
+};
+AnimatedContextMenu.propTypes = {
+  ...MenuPropTypes,
+};
```

Comment:
This is new. I could also do this in the index file instead and keep this file as just the animation container.